tracing-core/src/span.rs

```diff
@@ -208,34 +199,9 @@ impl<'a> Attributes<'a> {
     /// [record]: Attributes::record()
     /// [`Metadata`]: crate::metadata::Metadata
     /// [`FieldSet`]: crate::field::FieldSet
-    pub fn fields(&self) -> &FieldSet {
-        self.values.field_set()
-    }
-}
-
-// ===== impl Record =====
-
-impl<'a> Record<'a> {
-    /// Constructs a new `Record` from a `ValueSet`.
-    pub fn new(values: &'a field::ValueSet<'a>) -> Self {
-        Self { values }
-    }
-
-    /// Records all the fields in this `Record` with the provided [Visitor].
-    ///
-    /// [visitor]: super::field::Visit
-    pub fn record(&self, visitor: &mut dyn ValuableVisit) {
-        self.values.record(visitor)
-    }
-
-    /// Returns `true` if this `Record` contains a value for the given `Field`.
-    pub fn contains(&self, field: &field::Field) -> bool {
-        self.values.contains(field)
-    }
-
-    /// Returns true if this `Record` contains _no_ values.
-    pub fn is_empty(&self) -> bool {
-        self.values.is_empty()
+    pub fn fields(&self) -> &[valuable::NamedField<'_>] {
+        // TODO, this method doesn't exist yet:
+        self.values.fields()
     }
```

Comment:
We're going to need to add a `fields` method to `valuable::NamedValues`.

Reply:
can you make it a `todo!()` or something so we can merge this?